snuba/utils/streams/consumer.py

```diff
@@ -25,9 +26,22 @@
 logger = logging.getLogger(__name__)
 
 
-class TopicPartition(NamedTuple):
-    topic: str
-    partition: int
+@dataclass(frozen=True)
+class Topic:
+    __slots__ = ["name"]
+
+    name: str
+
+    def __contains__(self, partition: Partition) -> bool:
+        return partition.topic == self
```

Comment:
Was there a specific reason to switch from namedtuple to dataclass ?

Reply:
Mostly just consistency, and since `Topic` changed types from string there is less value in unpacking it as `(topic, partition)`.